docs/Inside_Arbitrum_Nitro/Inside_Arbitrum_Nitro.md

```diff
@@ -0,0 +1,527 @@
+- How does Arbitrum support Ethereum-compatible contracts and transactions?
+- How are ETH and tokens transferred into and out of Arbitrum chains, and how are they managed while on the chain?
+- How can I run my own Arbitrum node or validator?
+
+## Nitro's Design: The Four Big Ideas
+
+The essence of Nitro, and its key innovations, lie in four big ideas. We'll list them here with a very quick summary of each, then we'll unpack them in more detail in later sections.
+
+**Big Idea: Sequencing, Followed by Deterministic Execution**: Nitro processes transactions with a two-phase strategy. First, the transactions are organized into a single ordered sequence, and Nitro commits to that sequence. Then the transactions are processed, in that sequence, by a deterministic state transition function.
+
+**Big Idea: Geth at the Core**: Nitro supports Ethereum's data structures, formats, and virtual machine by compiling in the core code of the popular go-ethereum ("geth") Ethereum node software. Using geth as a library in this way ensures a very high degree of compatibility with Ethereum.
+
+**Big Idea: Separate Execution from Proving**: Nitro takes the same source code and compiles it twice, once to native code for execution in a Nitro node, optimized for speed, and again to WASM for use in proving, optimized for portability and security. 
+
+**Big Idea: Optimistic Rollup with Interactive Fraud Proofs**: Nitro settles transactions to the Layer 1 Ethereum chain using an optimistic rollup protocol, including the interactive fraud proofs pioneered by Arbitrum.
+
+## Sequencing, Followed by Deterministic Execution
+
+This diagram summarizes how transactions are processed in Nitro.
+
+![seq-then-exec](seq-then-exec.png)
+
+Let's follow a user's transaction through this process.
+
+First, the user creates a transaction, uses their wallet to sign it, and sends it to the Nitro chain's sequencer. The sequencer's job, as its name implies, is to take the arriving transactions, put them into an ordered sequence, and publish that sequence.
+
+Once the transactions are sequenced, they are run through the *state transition function*, one by one, in order. The state transition function takes as input the current state of the chain (account balances, contract code, and so on), along with the next transaction. It updates the state and sometimes emits a new Layer 2 block on the Nitro chain.
+
+Because the protocol doesn't trust the sequencer not to put garbage into its sequence, the state transition function will detect and discard any invalid transactions in the sequence. A well-behaved sequencer will filter out invalid transactions so the state transition function never sees them--and this reduces cost and therefore keeps transactions fees low--but Nitro will still work correctly no matter what the sequencer puts into its feed. (Transactions in the feed are signed by their senders, so the sequencer can't create forged transactions.)
```

Comment:
```suggestion
Because the protocol doesn't trust the sequencer not to put garbage into its sequence, the state transition function will detect and discard any invalid (i.e., improperly formed) transactions in the sequence. A well-behaved sequencer will filter out invalid transactions so the state transition function never sees them--and this reduces cost and therefore keeps transactions fees low--but Nitro will still work correctly no matter what the sequencer puts into its feed. (Transactions in the feed are signed by their senders, so the sequencer can't create forged transactions.)
```